### PR TITLE
{chem}[intel/2017a] libxc v3.0.0, ABINIT v8.2.2

### PR DESCRIPTION
**Pull request by migueldiascosta**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/l/libxc/libxc-3.0.0-intel-2017a.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+source_urls = ['http://www.tddft.org/programs/octopus/down.php?file=libxc/']
+
+configopts = 'FC="$F77" FCFLAGS="$FFLAGS" --enable-shared --enable-fortran'
+
+# From the libxc mailing list
+# To summarize: expect less tests to fail in libxc 2.0.2, but don't expect
+# a fully working testsuite soon (unless someone wants to volunteer to do
+# it, of course  ) In the meantime, unless the majority of the tests
+# fail, your build should be fine.
+# runtest = 'check'
```
Comment (wpoely86) [inline easybuild/easyconfigs/l/libxc/libxc-3.0.0-intel-2017a.eb]: is this still the case with 3.0.0?

Comment (migueldiascosta) [inline easybuild/easyconfigs/l/libxc/libxc-3.0.0-intel-2017a.eb]: this was already in the 3.0.0 easyconfigs for older toolchain versions, but you are right, I just uncommented `runtest` and it passes

File: easybuild/easyconfigs/l/libxc/libxc-3.0.0-intel-2017a.eb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+toolchain = {'name': 'intel', 'version': '2017a'}
+# Results for some functionals (e.g. mgga_c_tpss) deviate with too aggressive optimization settings.
+toolchainopts = {'lowopt': True}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['http://www.tddft.org/programs/octopus/down.php?file=libxc/']
+
+configopts = 'FC="$F77" FCFLAGS="$FFLAGS" --enable-shared --enable-fortran'
+
+runtest = 'check'
```
Comment (boegel) [inline easybuild/easyconfigs/l/libxc/libxc-3.0.0-intel-2017a.eb]: @migueldiascosta Can you open a separate PR to enable the tests in the existing libxc 3.0.0 easyconfigs?

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
node2592.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/bda00d842adb140eb2f86ee629b40266 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2065.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/4be7aa3e076003f93ead4a0197237ed3 for a full test report.

**Comment by boegel** [comment]
Hmm, tests are still failing for me on Intel Haswell, log is available at https://gist.github.com/boegel/0399ca3b97fe5fc0f52a947b864c58a8

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2507.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/ebe43ce551979d3c7f598920b0d284aa for a full test report.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
nscc04 - Linux RHEL 6.8, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/1efc26ec76ecbb772312d537af113333 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2428.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/38025ffebead6a98abc2730d28e80405 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2065.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/132866ed64cc376580fcad49647050ba for a full test report.

**Comment by boegel** [comment]
@migueldiascosta disabling `optarch` seems to fix the problem with the tests...

@wpoely86, @tovrstra Any idea on the potential (performance) impact of this, given that disabling `optarch` effectively disables the use of `-xHost` (cfr. http://easybuild.readthedocs.io/en/latest/Controlling_compiler_optimization_flags.html#default-behaviour).

**Comment by tovrstra** [comment]
It depends on the software that uses LibXC. E.g. in my own work, it just isn't the bottlenect. (Preparing input for LibXC and processing its output is the expensive part.) It may be different for ABINIT.

**Comment by migueldiascosta** [comment]
#4801 is correct, there is a typo in this and the easyconfig it was based on (ABINIT-8.2.2-intel-2016b.eb, already in master) that effectively disabled the external libxc and hid the fact that libc 3.x is not yet supported. Closing since this particular combination doesn't make sense, and it's probably better to treat libxc and ABINIT separately.

**Pull request closed without merging**